src/Directives.jl

```diff
@@ -95,11 +99,20 @@ function parsebracket end
 parsebracket(text :: AbstractString) = parsebracket(getdirective(text)...)
 parsebracket{D}(:: Directive{D}, text) = error("Unknown directive: '$D'")
 
+function parsebracket{D}(TD :: Directive{D}, arg, text)
+    if strip(arg) == ""
+        parsebracket(TD, text)
+    else
+        error("parsebracket not defined for directive '$D' with positional argument.")
+    end
+end
+
 function parsebracket(:: directive"docs", text)
+    eval(Expr(:toplevel, Expr(:using, :Lexicon)))
```

Comment:
I think the problem is that ``local_doc`` should be applied *after* the macro has returned. That should avoid the need to ``eval`` anything:

```julia
for each in split(text, "\n")
    s = strip(each)
    isempty(s) || push!(out, :(local_doc(@object($parse(s)))))
end
```

Where ``@object`` does the first part of documentation lookup, ie. find the object represented by the expression ``ex`` but does not then wrap it in an ``@doc``:

```julia
macro object(ex)
    if isexpr(ex, :call)
        name = esc(namify(ex.args[1]))
        if any(x -> isexpr(x, :(::)), ex.args)
            :(methods($(name), $(esc(signature(ex))))[1])
        else
            :(@which($esc(ex)))
        end
    elseif isvar(ex)
        :(@var($(esc(ex))))
    else
        esc(ex)
    end
end
```

``local_doc`` will then have access to the actual object rather than the parsed expression which should make things much simpler to deal with.

I've not properly tested whether this code above actually does work for all possible cases, but it should in principle. I'll commit the ``@object`` macro to the branch shortly.